### PR TITLE
Create default network when provisioning project

### LINKED ISSUE #21: [coldfront] Create default network for provisioned projects
As a user, I don't want to have to worry about the complexities of creating my first network when starting a project. I want to have one set up for me.

### PR DESCRIPTION
**Pull request by knikolla**
- Added attribute 'OpenStack Public Network ID'
- Added attribute 'OpenStack Default Network CIDR'
- Default network DNS set to Google DNS
- Added simple test case of network created

Closes #21

### CODE REVIEW

**Comment by knikolla** [comment]
@larsks would appreciate a review from you on this one.

File: src/coldfront_plugin_openstack/tasks.py [hunk 1]
```diff
@@ -137,6 +139,92 @@ def assign_role_on_user(resource, username, project_id):
...
+            }
+        }
+        network = neutron.create_network(body=default_network_payload)
+        logger.info(f'Default network with ID {network["network"]["id"]} '
+                    f'created for project {project_id}.')
+
+    # Get or create default subnet
+    subnets = neutron.list_subnets(project_id=project_id,
+                                   name='default_subnet')
+    if subnets := subnets['subnets']:
```
Comment (larsks) [inline src/coldfront_plugin_openstack/tasks.py:168]: Do you need to worry about the situation in which a `default_subnet` exists but it is associated with something other than `default_network`?

**Review by larsks — APPROVED** [review]
Looks good to me, although I left one question about the degree of error checking which is appropriate for this code.

File: src/coldfront_plugin_openstack/tasks.py (hunk 1 above)
Comment (knikolla) [reply src/coldfront_plugin_openstack/tasks.py:168]: Good question. Given that this code will only run on new project creations, I don't think we do need to worry about that. This error checking is mostly to protect against half-executions of itself and ensure a certain degree of idempotency.

File: src/coldfront_plugin_openstack/tests/base.py [hunk 2]
```diff
@@ -103,6 +105,12 @@ def new_resource(name=None, auth_url=None) -> Resource:
             resource=openstack,
             value=1
         )
+        ResourceAttribute.objects.get_or_create(
+            resource_attribute_type=ResourceAttributeType.objects.get(
+                name=attributes.RESOURCE_DEFAULT_PUBLIC_NETWORK),
+            resource=openstack,
+            value=os.getenv('OPENSTACK_PUBLIC_NETWORK_ID')
```
Comment (jtriley) [inline src/coldfront_plugin_openstack/tests/base.py:112]: Should this be parsed similar to `OPENSTACK_DEVSTACK_APPLICATION_CREDENTIAL_ID`? For example if there are multiple openstack clusters...

**Comment by jtriley** [comment]
Need up update the README with these new options.

**Pull request merged**